source/whereAny.js

```diff
@@ -0,0 +1,52 @@
+import _curry2 from './internal/_curry2';
+import _has from './internal/_has';
+import _isEmpty from './isEmpty';
+
+
+/**
+ * Takes a spec object and a test object; returns true if the test satisfies
+ * the spec. Each of the spec's own properties must be a predicate function.
+ * Each predicate is applied to the value of the corresponding property of the
+ * test object. `whereAny` returns true if at least one of the predicates return true, false
+ * otherwise.
+ *
+ * `whereAny` is well suited to declaratively expressing constraints for other
+ * functions such as [`filter`](#filter) and [`find`](#find).
+ *
+ * @func
+ * @memberOf R
+ * @since v0.27.0
+ * @category Object
+ * @sig {String: (* -> Boolean)} -> {String: *} -> Boolean
+ * @param {Object} spec
+ * @param {Object} testObj
+ * @return {Boolean}
+ * @see R.propSatisfies, R.where
+ * @example
+ *
+ *      // pred :: Object -> Boolean
+ *      const pred = R.whereAny({
+ *        a: R.equals('foo'),
+ *        b: R.complement(R.equals('xxx')),
+ *        x: R.gt(R.__, 10),
+ *        y: R.lt(R.__, 20)
+ *      });
+ *
+ *      pred({a: 'foo', b: 'xxx', x: 8, y: 34}); //=> true
+ *      pred({a: 'xxx', b: 'xxx', x: 9, y: 21}); //=> false
+ *      pred({a: 'bar', b: 'xxx', x: 10, y: 20}); //=> false
+ *      pred({a: 'foo', b: 'bar', x: 10, y: 20}); //=> true
+ *      pred({a: 'foo', b: 'xxx', x: 11, y: 20}); //=> true
+ */
+var whereAny = _curry2(function where(spec, testObj) {
+  if (_isEmpty(spec)) {
+    return true;
```

Comment:
I don't think so.  This is a disjunction, which means that the vacuous case should default to `false`, no?

That is, where `all` could be written as `reduce(and, true)`, this is more like `any`, which is `reduce(or, false)`.  It asks the question "Do any of these properties hold?"  If there are no properties, it's false.

Reply:
i can agree with this, makes sense...

I took this case from the [`where spec`](https://github.com/ramda/ramda/blob/a52431a9d24f272d220fba8daf2d03c74248c474/test/where.js#L40:42)

Happy to change it

Reply:
Sounds good.